src/libaktualizr/package_manager/ostreemanager.cc

```diff
@@ -18,13 +18,7 @@
 
 static void aktualizr_progress_cb(OstreeAsyncProgress *progress, gpointer data) {
   auto *mt = static_cast<PullMetaStruct *>(data);
-  bool has_been_paused = false;
-  if (!mt->pause_mutex->try_lock()) {  // If can't aquire the mutex thats mean
-    mt->pause_mutex->lock();           // mutex is locked by pause thread, and
-    has_been_paused = true;            // we will wait, until it is locked.
-  } else {
-    mt->pause_mutex->unlock();  // Unlock locked by try_lock() mutex.
-  }
+  mt->pause();
```

Comment:
Can we add a comment here, to document that we expect to wait for the download to not be paused anymore?

This callback is passed through several method calls and constructors and the intent is a bit obfuscated here, IMO.

Reply:
Agree, also the name is a bit confusing, as it will not pause on every call, but I wasn't able to come up with something better. I'll add a comment.

Reply:
+1 for a comment, but even just `checkPause` would be a better name.

Reply:
True, I'll rename it.

Reply:
Sorry, a bit late again. But it probably wouldn't hurt to have a null check as well before making the call.

Reply:
@lbonn I don't know, std:function will throw in case it's empty. Do we want to proceed if pause callback isn't provided?

Reply:
Just having:

```
if (checkPause != nullptr) {
  checkPause()
}
```

Then, users of the class could just pass a null pointer if they don't need the pause functionality. It's only an added functionality, not a part of the core duty of the class.

Reply:
Agreed, that would make sense.